api/spec/src/types.tsp

```diff
@@ -44,6 +44,15 @@ scalar DateTime extends utcDateTime;
 @example("0.0015")
 scalar Money extends string;
 
+/**
+ * Helper types to turn the type information (commonly used at creat and update scenarios) into resources.
+ *
+ * This way `Resource` doesn't have to be extended and the underlying type can be reused freely.
+ */
+model AsResource<T> extends Resource {
+  ...T;
+}
```

Comment:
It's not used anywhere as far as I see.

```suggestion
/**
 * Helper types to turn the type information (commonly used at creat and update scenarios) into resources.
 *
 * This way `Resource` doesn't have to be extended and the underlying type can be reused freely.
 */
model AsResource<T> extends Resource {
  ...T;
}
```

Reply:
Yes, thanks, i'll delete it

I think we should talk about the Resource model separately: Its not what our resources look like so in my opinion it's quite misleading.